Frameworks/Foundation/NSString.mm

```diff
@@ -680,7 +680,25 @@ - (instancetype)pathExtension {
  @Status Interoperable
 */
 - (instancetype)lastPathComponent {
-    return [self substringFromIndex:_CFStartOfLastPathComponent2(static_cast<CFStringRef>(self))];
+    if ([self length] == 0) {
+        return @"";
+    }
+
+    CFIndex index = _CFStartOfLastPathComponent2(static_cast<CFStringRef>(self));
```

Comment:
>_CFStartOfLastPathComponent2 [](start = 20, length = 28)

_CFStartOfLastPathComponent2 will not be helpful in case of trailing slashes as this would end up pointing to the slash before the last.